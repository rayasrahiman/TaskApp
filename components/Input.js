import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  Platform,
  TouchableOpacity,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import AntIcon from 'react-native-vector-icons/AntDesign';

import {Colors} from '../constants/colors';
import I18n from '../languages/i18n';

export default function Input({
  label,
  placeholder,
  titleContProp,
  titleHeaderProp,
  txtInputContProp,
  txtInputProp,
  onChangeText,
  iconName,
  error,
  password,
  inputValue,
  onFocus = () => {},
  ...inputProps
}) {
  const [isFocused, setIsFocused] = useState(false);
  const [hidePass, setHidePass] = useState(password);

  return (
    <View>
      <View style={{...styles.titleCont, ...titleContProp}}>
        <Text style={{...styles.titleHeader, ...titleHeaderProp}}>{label}</Text>
      </View>
      <View
        style={{
          ...styles.txtInputCont,
          ...{
            borderColor: error
              ? Colors.error
              : isFocused
              ? Colors.primary700
              : Colors.gray700,
            backgroundColor: inputValue ? '#888' : Colors.backGround,
          },
          ...txtInputContProp,
        }}>
        <TextInput
          {...inputProps}
          style={{
            ...styles.txtInput,
            ...{color: inputValue ? Colors.gray700 : Colors.primary700},
            ...txtInputProp,
          }}
          autoCorrect={false}
          onFocus={() => {
            onFocus();
            setIsFocused(true);
          }}
          onBlur={() => {
            setIsFocused(false);
          }}
          placeholder={placeholder}
          onChangeText={onChangeText}
          secureTextEntry={hidePass}
        />
        {password && (
          <View style={{flexDirection: 'row'}}>
            {error && (
              <AntIcon
                name="exclamationcircle"
                style={[
                  styles.iconSize,
                  {marginRight: wp(2), color: Colors.error},
                ]}
              />
            )}
            <Icon
              onPress={() => setHidePass(!hidePass)}
              name={hidePass ? 'eye-off-outline' : 'eye-outline'}
              style={styles.iconSize}
            />
          </View>
        )}
      </View>
      {error && (
        <View style={{...styles.titleCont, ...{marginTop: hp(0.2)}}}>
          <Text style={styles.errTxt}>{error}</Text>
        </View>
      )}
      {password && (
        <TouchableOpacity style={styles.titleCont}>
          <Text style={styles.txtInput}>{I18n.t('ForgotPassword')}</Text>
        </TouchableOpacity>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  titleCont: {
    marginHorizontal: Platform.OS === 'android' ? wp(5) : wp(3),
  },
  titleHeader: {
    fontWeight: 'bold',
    fontSize: Platform.OS === 'android' ? hp(3) : hp(2),
  },
  txtInputCont: {
    marginVertical: 10,
    marginHorizontal: Platform.OS === 'android' ? wp(5) : wp(3),
    height: Platform.OS === 'android' ? hp(7) : hp(6),
    flexDirection: 'row',
    paddingHorizontal: Platform.OS === 'android' ? wp(4) : wp(2),
    borderWidth: wp(0.3),
    borderRadius: hp(1),
    // width: '100%',
    borderColor: Colors.gray700,
    alignItems: 'center',
  },
  txtInput: {
    color: Colors.primary700,
    flex: 1,
    fontSize: Platform.OS === 'android' ? hp(3) : hp(2),
  },
  iconSize: {
    fontSize: hp(3),
  },
  errTxt: {
    color: Colors.error,
    fontSize: hp(3),
  },
});
