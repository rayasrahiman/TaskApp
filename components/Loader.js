import {View, ActivityIndicator, StyleSheet} from 'react-native';
import React from 'react';
import {heightPercentageToDP as hp} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';

export default function Loader() {
  return (
    <View style={styles.container}>
      <ActivityIndicator color={Colors.primary700} size={hp(4)} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
