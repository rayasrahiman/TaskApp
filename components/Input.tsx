import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  Platform,
  TouchableOpacity,
} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import AntIcon from 'react-native-vector-icons/AntDesign';

import {Colors} from '../constants/colors';
import I18n from '../languages/i18n';
import Loader from './Loader';

export type Props = {
  label: string;
  placeholder: string;
  titleContProp: any;
  titleHeaderProp: any;
  txtInputContProp: any;
  txtInputProp: any;
  onChangeText: (text: string) => void;
  iconName: string;
  txt: string;
  error: boolean;
  password: boolean;
  forgotPassword: boolean;
  passwordLoader: boolean;
  createPass: boolean;
  inputValue: boolean;
  confirmNew: boolean;
  onFocus(): void;
  forgotPass(): void;
};

const Input: React.FC<Props> = ({
  label,
  placeholder,
  titleContProp,
  titleHeaderProp,
  txtInputContProp,
  txtInputProp,
  onChangeText,
  iconName,
  error,
  password,
  forgotPassword,
  passwordLoader,
  inputValue,
  txt,
  onFocus,
  forgotPass,
  createPass,
  confirmNew,
  ...inputProps
}) => {
  const [isFocused, setIsFocused] = useState(false);
  const [hidePass, setHidePass] = useState(password);

  return (
    <View>
      <View style={{...styles.titleCont, ...titleContProp}}>
        <Text style={{...styles.titleHeader, ...titleHeaderProp}}>{label}</Text>
      </View>
      <View
        style={{
          ...styles.txtInputCont,
          ...{
            borderColor: error
              ? Colors.red
              : isFocused
              ? Colors.black
              : Colors.grayishBlack,
            backgroundColor: inputValue ? Colors.lightGray : Colors.white,
          },
          ...txtInputContProp,
        }}>
        <TextInput
          {...inputProps}
          style={{
            ...styles.txtInput,
            ...{color: inputValue ? Colors.grayishBlack : Colors.black},
            ...txtInputProp,
          }}
          autoCorrect={false}
          onFocus={() => {
            onFocus();
            setIsFocused(true);
          }}
          onBlur={() => {
            setIsFocused(false);
          }}
          placeholder={placeholder}
          onChangeText={onChangeText}
          secureTextEntry={hidePass}
        />
        {password && (
          <View style={{flexDirection: 'row'}}>
            {error && (
              <AntIcon
                name="exclamationcircle"
                style={[
                  styles.iconSize,
                  {marginRight: wp(2), color: Colors.error},
                ]}
              />
            )}
            {confirmNew && (
              <AntIcon
                name="checkcircle"
                style={[
                  styles.iconSize,
                  {marginRight: wp(2), color: Colors.green},
                ]}
              />
            )}
            <Icon
              onPress={() => setHidePass(!hidePass)}
              name={hidePass ? 'eye-off-outline' : 'eye-outline'}
              style={styles.iconSize}
            />
          </View>
        )}
      </View>
      {error && (
        <View style={{...styles.titleCont, ...{marginTop: hp(0.2)}}}>
          <Text style={styles.errTxt}>{error}</Text>
        </View>
      )}
      {createPass && (
        <View style={{...styles.titleCont, ...{marginVertical: hp(3)}}}>
          <View style={styles.createPassCont}>
            <Icon
              name="check"
              style={{
                ...styles.iconSize,
                ...{
                  color: txt.match(/(?=^.{8,}$)/) ? Colors.green : Colors.gray,
                  marginRight: wp(1),
                },
              }}
            />
            <Text style={styles.createPassTxt}>
              {I18n.t('eightCharacters')}
            </Text>
          </View>
          <View style={styles.createPassCont}>
            <Icon
              name="check"
              style={{
                ...styles.iconSize,
                ...{
                  color: txt.match(/(?=.*[a-z])(?=.*[A-Z])/)
                    ? Colors.green
                    : Colors.gray,
                  marginRight: wp(1),
                },
              }}
            />
            <Text style={styles.createPassTxt}>
              {I18n.t('upperandlowercase')}
            </Text>
          </View>
          <View style={styles.createPassCont}>
            <Icon
              name="check"
              style={{
                ...styles.iconSize,
                ...{
                  color: txt.match(/(?=.*[0-9])/) ? Colors.green : Colors.gray,
                  marginRight: wp(1),
                },
              }}
            />
            <Text style={styles.createPassTxt}>{I18n.t('oneNumber')}</Text>
          </View>
          <View style={styles.createPassCont}>
            <Icon
              name="check"
              style={{
                ...styles.iconSize,
                ...{
                  color: txt.match(
                    /(?=.*[~`!@#$%^&*()--+={}\[\]|\\:;"'<>,.?/_â‚¹])/,
                  )
                    ? Colors.green
                    : Colors.gray,
                  marginRight: wp(1),
                },
              }}
            />
            <Text style={styles.createPassTxt}>{I18n.t('specialChar')}</Text>
          </View>
        </View>
      )}
      {forgotPassword && (
        <TouchableOpacity style={styles.titleCont} onPress={forgotPass}>
          <Text style={styles.txtInput}>{I18n.t('ForgotPassword')}</Text>
        </TouchableOpacity>
      )}
      {passwordLoader && <Loader />}
    </View>
  );
};

const styles = StyleSheet.create({
  titleCont: {
    marginHorizontal: Platform.OS === 'android' ? wp(5) : wp(3),
  },
  titleHeader: {
    fontWeight: 'bold',
    fontSize: Platform.OS === 'android' ? hp(3) : hp(2),
    color: Colors.black,
  },
  txtInputCont: {
    marginVertical: 10,
    marginHorizontal: Platform.OS === 'android' ? wp(5) : wp(3),
    height: Platform.OS === 'android' ? hp(7) : hp(6),
    flexDirection: 'row',
    paddingHorizontal: Platform.OS === 'android' ? wp(4) : wp(2),
    borderWidth: wp(0.3),
    borderRadius: hp(1),
    // width: '100%',
    borderColor: Colors.grayishBlack,
    alignItems: 'center',
  },
  txtInput: {
    color: Colors.darkBlue,
    flex: 1,
    fontSize: Platform.OS === 'android' ? hp(3) : hp(2),
  },
  iconSize: {
    fontSize: hp(3),
  },
  errTxt: {
    color: Colors.error,
    fontSize: hp(3),
  },
  createPassCont: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  createPassTxt: {
    fontSize: hp(3),
    color: Colors.grayishBlack,
  },
});

export default Input;
