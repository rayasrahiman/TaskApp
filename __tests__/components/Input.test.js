import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import Input from '../../components/Input';
import Loader from '../../components/Loader';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('Input', () => {
  test('should render Input correctly', () => {
    const wrapper = render(<Input />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render Loader', () => {
    const wrapper = shallow(<Input />);
    wrapper.find(<Loader />);
  });
});
