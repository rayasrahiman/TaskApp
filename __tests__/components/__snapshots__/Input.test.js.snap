// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input correctly 1`] = `
<View>
  <View
    style={
      Object {
        "marginHorizontal": 22.5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 26.5,
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#838383",
        "borderRadius": 13.5,
        "borderWidth": 2.5,
        "flexDirection": "row",
        "height": 80,
        "marginHorizontal": 22.5,
        "marginVertical": 10,
        "paddingHorizontal": 15,
      }
    }
  >
    <TextInput
      autoCorrect={false}
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 26.5,
        }
      }
    />
  </View>
</View>
`;
